beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/AbstractBlockPublisher.java

```diff
@@ -75,7 +77,18 @@ public SafeFuture<SendSignedBlockResult> sendSignedBlock(
       final BlockPublishingPerformance blockPublishingPerformance) {
     return blockFactory
         .unblindSignedBlockIfBlinded(blockContainer.getSignedBlock(), blockPublishingPerformance)
-        .thenPeek(performanceTracker::saveProducedBlock)
+        .whenComplete(
+            (block, ex) -> {
+              if (ex != null) {
+                // in case of relay API timeouts when unblinding, we still want to assume the block
+                // as produced, since the relay could have published it
+                if (ExceptionUtil.hasCause(ex, TimeoutException.class)) {
+                  performanceTracker.saveProducedBlock(blockContainer.getSignedBlock());
+                }
+              } else {
+                performanceTracker.saveProducedBlock(block);
```

Comment:
do we really need to discriminate the exception here? wandering if we can save the block as produced

Reply:
It doesn't hurt since we filter late compared to the chain we have anyways and exceptions shouldn't really happen

Reply:
but you could have a bunch of errors on API side, not just a timeout, and you never know if on the other side it got the info to publish.
Moreover, I think that we know we attempted to publish a produced block, so it make sense to me to blindly track it at the very beginning of the flow...

Reply:
Committed a change to always save the block as produced